Troupon.Checkout.Api/Startup.cs

```diff
@@ -58,46 +54,45 @@ public class Startup
         Assembly.GetExecutingAssembly()
           .GetName()
           .Name);
-
+      
       services.AddEfReadRepository<CheckoutDbContext>();
       services.AddEfWriteRepository<CheckoutDbContext>();
+      //services.AddOpenApi(AuthSettings);
       services.AddControllers();
       services.AddOpenApi(Configuration);
       services.AddMetrics();
       services.AddFluentValidaton();
-      services.AddMemoryCache();
+      services.AddMemoryCache();      
+
+      services.Configure<MvcOptions>(o =>
+      {
+        o.Filters.Add(new ProducesAttribute("application/json", "application/xml"));
+        o.Filters.Add(new ConsumesAttribute("application/json", "application/xml"));
+      });
     }
 
     // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
     public void Configure(
       IApplicationBuilder app,
+      IApiVersionDescriptionProvider apiVersionDescriptionProvider,
       IWebHostEnvironment env,
       IDbContextFactory<CheckoutDbContext> dbContextFactory)
-    {
-      //if (env.IsDevelopment())
-      //{
-      //    app.UseDeveloperExceptionPage();
-      //}
+    {      
       app.UseExceptionHandler("/error");
 
       app.UseHttpsRedirection();
       app.UseSerilogRequestLogging();
-
-      // app.UsePathBase("/graphql");
-
-      //catalogDbContext.Database.EnsureDeleted();
-      var catalogDbContext = dbContextFactory.CreateDbContext();
-      catalogDbContext.Database.Migrate();
-
-      // app.UsePlayground();
-
-      app.UseSwagger();
+     
+      app.UseSwagger();      
       app.UseSwaggerUI(
         c =>
         {
-          c.SwaggerEndpoint(
-            "/swagger/v1/swagger.json",
-            "Troupon Checkout");
+          foreach (var description in apiVersionDescriptionProvider.ApiVersionDescriptions)
+          {
+            c.SwaggerEndpoint($"/swagger/{description.ApiVersion}/swagger.json", $"Troupon Checkout Specification{description.ApiVersion}");
+            c.RoutePrefix = string.Empty;
+          }
+
           c.RoutePrefix = string.Empty;
```

Comment:
Should this line be removed since it's already inside foreach?

Reply:
Done!